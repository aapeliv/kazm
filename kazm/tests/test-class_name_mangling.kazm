/* Make sure that we can have multiple classes with methods of the same name */

class Class1 {
  int x;
  bool y;
  int z;

  void do_something() {
    int_println(me.x);
  }
};

class Class2 {
  int x;
  int y;
  bool z;

  void do_something() {
    println("Hello from Class2");
  }
};

int main() {
  Class1 c1;
  c1.x = 12;
  Class2 c2;
  c2.do_something();
  c1.do_something();
}
