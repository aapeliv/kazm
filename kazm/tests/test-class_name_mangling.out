Hello from Class2
12
